jvb/src/main/kotlin/org/jitsi/videobridge/cc/allocation/PacketHandler.kt

```diff
@@ -0,0 +1,201 @@
+     */
+    fun accept(packetInfo: PacketInfo): Boolean {
+        val videoPacket = packetInfo.packetAs<VideoRtpPacket>()
+        val adaptiveSourceProjection = adaptiveSourceProjectionMap[videoPacket.ssrc]
+        if (adaptiveSourceProjection == null) {
+            logger.debug { "Dropping an RTP packet for an unknown SSRC: ${videoPacket.ssrc}" }
+            numDroppedPacketsUnknownSsrc.incrementAndGet()
+            return false
+        }
+        return adaptiveSourceProjection.accept(packetInfo)
+    }
+
+    /**
+     * @return true if [rtcpSrPacket] should be accepted, false otherwise.
+     *
+     * Filters out packets that match one of the streams that this instance manages, but don't match the target SSRC.
+     * Allows packets for streams not managed by this instance.
+     */
+    fun accept(rtcpSrPacket: RtcpSrPacket): Boolean {
+        val adaptiveSourceProjection = adaptiveSourceProjectionMap[rtcpSrPacket.senderSsrc]
+
+        // This is probably for an audio stream. In any case, if it's for a stream which we are not forwarding it
+        // will be stripped off at a later stage (in RtcpSrUpdater).
+        return adaptiveSourceProjection == null ||
+            // We only accept SRs for the SSRC that we're forwarding with.
+            adaptiveSourceProjection.targetSsrc == rtcpSrPacket.senderSsrc
+    }
+
+    /**
+     * @return true if the packet was transformed successfully, false otherwise.
+     */
+    fun transformRtcp(rtcpSrPacket: RtcpSrPacket): Boolean =
+        adaptiveSourceProjectionMap[rtcpSrPacket.senderSsrc]?.rewriteRtcp(rtcpSrPacket) ?: false
+
+    /**
+     * Utility method that looks-up or creates the adaptive source projection of
+     * a source.
+     */
+    private fun lookupOrCreateAdaptiveSourceProjection(singleAllocation: SingleAllocation): AdaptiveSourceProjection? {
+        val source = singleAllocation.mediaSource
+        val endpointID = singleAllocation.endpointId
+        if (source == null) {
+            return null
+        }
+        synchronized(adaptiveSourceProjectionMap) {
+            var adaptiveSourceProjection = adaptiveSourceProjectionMap[source.primarySSRC]
+            if (adaptiveSourceProjection != null) {
+                return adaptiveSourceProjection
+            }
+            val rtpEncodings = source.rtpEncodings
+            if (ArrayUtils.isNullOrEmpty(rtpEncodings)) {
+                return null
+            }
+
+            // XXX the lambda keeps a reference to the sourceBitrateAllocation
+            // (a short lived object under normal circumstances) which keeps
+            // a reference to the Endpoint object that it refers to. That
+            // can cause excessive object retention (i.e. the endpoint is expired
+            // but a reference persists in the adaptiveSourceProjectionMap). We're
+            // creating local final variables and pass that to the lambda function
```

Comment:
"final variables" is the Java idiom, in Kotlin they're `val`s.

Also the grammar of this comment is a bit wonky, though that's not new from the Java version.

Reply:
The comment and use of a final variable is no longer valid (we only pass a MediaSourceDesc which doesn't have back references to Endpoint), so I removed it.